src/hupper/ipc.py

```diff
@@ -184,10 +180,6 @@ def send(self, value):
             self._write_packet(data)
         return len(data) + self._packet_len.size
 
-    def recv(self, timeout=None):
-        packet = self.reader_queue.get(block=True, timeout=timeout)
-        return packet
```

Comment:
got rid of the reader queue here and instead just invoke a callback letting the owner of the pipe handle packets how they wish (using a deque, notify the control pipe, etc)